lib/rubygems/specification.rb

```diff
@@ -1667,7 +1667,7 @@ def has_conflicts?
   # https://reproducible-builds.org/specs/source-date-epoch/
 
   def date
-    @date ||= ENV["SOURCE_DATE_EPOCH"] ? Time.utc(*Time.at(ENV["SOURCE_DATE_EPOCH"].to_i).utc.to_a[3..5].reverse) : TODAY
+    @date ||= ENV["SOURCE_DATE_EPOCH"] ? Time.utc(*Gem.source_date_epoch.utc.to_a[3..5].reverse) : TODAY
```

Comment:
Shouldn't this be:
```ruby
@date ||= Gem.source_date_epoch ? Time.utc(*Gem.source_date_epoch.to_a[3..5].reverse) : TODAY
```

Reply:
No, because `Gem.source_date_epoch` always returns a value.

However, since `Gem.source_date_epoch` always returns a value, I think that means it _can_ become this:

```ruby
@date ||= Time.utc(*Gem.source_date_epoch.utc.to_a[3..5].reverse)
```

Reply:
It can probably be able to be simplified more than that, but I'd need to dig into the actual result of that function call to determine that.

Reply:
It looks like what that code does is take a UTC time and return that date, but at midnight UTC. So I can't think of a way off-hand to simplify it farther than what I said above, but that's still a nice improvement.

```
>> Time.now.utc
=> 2019-08-20 20:44:18 UTC
>> Time.utc(*Time.now.utc.to_a[3..5].reverse)
=> 2019-08-20 00:00:00 UTC
```